proxmox/pools/pool_types.go

```diff
@@ -6,6 +6,8 @@
 
 package pools
 
+import "github.com/bpg/terraform-provider-proxmox/internal/types"
```

Comment:
Oh... I didn't realize this mistake that I've made before -- the `proxmox` package (API client) is now dependent on the `internal` package types 😞
You're also adding new API types and more dependencies between API and FWK types in #498. 

This all won't work long term, as I think it will make sense to extract the API client into a separate repo (#423)

But I'll leave it as is for now, the "separation" refactoring can be done later.